aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -168,32 +169,29 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
-static Tensor sum_scan_exclusive(const Tensor& x, int64_t dim) {
-  Tensor ret = at::cumsum(-x, dim);
-
-  int64_t end_idx = ret.size(dim) - 1;
-  Tensor ret_sum = ret.narrow(dim, end_idx, 1).clone(at::MemoryFormat::Preserve);
-  ret -= ret_sum.expand_as(ret);
-  ret += x;
-  return ret;
+Tensor reversed_cumsum(const Tensor& w, int64_t dim) {
+  /* Implements w.flip(dim).cumsum(dim).flip(dim) without copying.
```

Comment:
nit - Logically implements w.flip(dim).cumsum(dim).flip(dim) ...

(to avoid confusion since we aren't actually using flip operation in the computation)